### PR TITLE
kernel: add kmod-leds-uleds

### PR DESCRIPTION
**Pull request by ktgeek**
The allows userspace LEDs to be created and controlled. This can be useful
for testing triggers and can also be used to implement virtual LEDs.

Signed-off-by: Keith T. Garner <kgarner@kgarner.com>

### CODE REVIEW

**Comment by aparcar** [comment]
Looks good to me.

**Comment by chunkeey** [comment]
@aparcar yes? Have you tested it?

Of course, userspace drivers need some userspace tools to be useful. But then, OpenWrt has had supported for other userspace drivers or testing drivers (FUSE (this is a big one), iio, spidev, userspace thermal/speed govenor, mac80211's hwsim). So it would fit right in!

So, if it works. Why not? :+1:

**Comment by ktgeek** [comment]
> Of course, userspace drivers need some userspace tools to be useful. But then, OpenWrt has had supported for other userspace drivers or testing drivers (FUSE (this is a big one), iio, spidev, userspace thermal/speed govenor, mac80211's hwsim). So it would fit right in!

I have a ["blinking lights" project](https://github.com/ktgeek/blinkstick-users) that is [almost ready](https://github.com/ktgeek/ktg-openwrt-packages/tree/master/bluld) to be submitted to misc packages. I've been using both of these for over a year at home, my thought was might as well make me easier for others to consume.

File: package/kernel/linux/modules/leds.mk [hunk 1]
```diff
@@ -145,3 +145,17 @@ define KernelPackage/leds-pwm/description
...
+define KernelPackage/leds-uleds
+  SUBMENU:=$(LEDS_MENU)
+  TITLE:=Userspace LEDs
+  KCONFIG:=CONFIG_LEDS_USER
+  FILES:=$(LINUX_DIR)/drivers/leds/uleds.ko
+  AUTOLOAD:=$(call AutoLoad,60,leds-uleds,1)
+endef
+
+define KernelPackage/leds-uleds/description
+ The options enables support for userspace LEDs.
```
Comment (aparcar) [inline package/kernel/linux/modules/leds.mk:158]:
```suggestion
 The option enables support for userspace LEDs.
```

**Review by aparcar — CHANGES_REQUESTED** [review]

File: package/kernel/linux/modules/leds.mk (hunk 1 above)
Comment (ktgeek) [reply package/kernel/linux/modules/leds.mk]: Thanks for that. 🤦

**Comment by aparcar** [comment]
Merged, thanks.

**Pull request closed without merging**